### DISCUSSION TITLE
Oauth integration

### DISCUSSION THREAD

**Discussion by matejko94** [fastapi-users/fastapi-users#1179] [General]
I am integration Oauth by example writen in docs. But than afther that whatevery I try reset password and verify account doesnt work ? So it is not possible to have auath and clasical login ? @frankie567 

My question is how to handle both Oauth and clasical login ?

**Comment by frankie567** [comment]
When a user registers with any OAuth provider (like Google, Facebook...), they are created in database with their email and a generated password. So reset password and verify routes should work as expected. Are you sure you tried with the right email?

**Comment by matejko94** [comment]
@frankie567  Thank you for prompt response. 

I had a problem with verification beacuse of wrongly integrating into alchemy. So now I integrated it like here in this example:
https://github.com/justb4/fastapi-users-starter/blob/main/alembic/versions/787703593e67_create_fastapi_users_user_table.py
The key thing is for alembic versioning:
`from fastapi_users_db_sqlalchemy import GUID`
So this is why verification didnt work form me. I change everything to int from GUID and there was issue.

I have some other question how is the easiest think to integrate facbook and gmail do I need 2 tables. I cannot find into documentaiton if I need 2 tables or is enough just 1 ?

**Reply by frankie567** [reply]
Ok!

One table is enough. FastAPI Users takes care of storing accounts in the same table and keep a reference to the provider in a dedicated column. You can see it here: https://github.com/fastapi-users/fastapi-users-db-sqlalchemy/blob/9a036bde2194aa2bcaa6b0d99eaf2a02571dcde9/fastapi_users_db_sqlalchemy/__init__.py#L68-L70

**Comment by matejko94** [comment]
@frankie567 

So here I have to make new endpoints ?
```
app.include_router(
    fastapi_users.get_oauth_router(google_oauth_client, auth_backend, SECRET),
    prefix="/auth/google",
    tags=["auth"],
)
```

**Reply by frankie567** [reply]
Exactly, one router per provider.

**Comment by matejko94** [comment]
Thank you. Btw very glad, that I found this auth provider for fastapi.

**Comment by matejko94** [comment]
@frankie567 if you have idea what we are doing wrong: https://stackoverflow.com/questions/75888521/google-oauth-integration-with-fastapi I really appritiate your time.
